### PR TITLE
Add execute metrics

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
Execution is occasionally slow but cause is not captured in current metrics

#### Summary of Changes
Add more timings 

Fixes #

### CODE REVIEW

**Comment by t-nelson** [comment]
@Lichtso @jackcmay in the spirit of this commit, are there any other metrics you would recommend adding to try to get better visibility into the `replay-slot-stats.execute_ms` metric contributors?  We're trying to suss out inconsistencies between replay time and compute costs, especially between successful and failed transactions

Wondering if anything below here in the calls stack might be interesting?
https://github.com/solana-labs/solana/blob/66fa8f966753b732facbbd66ff5b549be796ae3c/programs/bpf_loader/src/lib.rs#L1017-L1021

File: core/src/progress_map.rs [hunk 1]
```diff
@@ -121,6 +121,68 @@ impl ReplaySlotStats {
...
+                self.execute_timings
+                    .execute_acessories
+                    .refcells_to_accounts_us,
+                i64
+            ),
+            (
+                "update_executors_us",
+                self.execute_timings.execute_acessories.update_executors_us,
+                i64
+            ),
```
Comment (t-nelson) [inline core/src/progress_map.rs:185]: wdyt about giving these a common prefix?

**Review by t-nelson — DISMISSED** [review]
lgtm! bonus points for time units in the field names!

let's give @Lichtso and @jackcmay a bit to comment re what else might be good to add here

**Comment by Lichtso** [comment]
`create_executor()` needs a separate timing (as it contains the JIT compilation):
https://github.com/solana-labs/solana/blob/66fa8f966753b732facbbd66ff5b549be796ae3c/programs/bpf_loader/src/lib.rs#L80

**Comment by carllin** [comment]
@Lichtso great suggestion, added here: https://github.com/solana-labs/solana/pull/22043/commits/9f0d6e7505d19b6091305362fc695306d56c6f5f

Do you think it's necessary to break down the `execute_time` here: https://github.com/solana-labs/solana/blob/9c5d82557a965cf23831a3fb2e3b8986d379ce04/programs/bpf_loader/src/lib.rs#L1013-L1020 that covers the program execution into any further detail, i.e. more granularity in `execute_program_jit`?

**Comment by Lichtso** [comment]
> @Lichtso great suggestion, added here: [1da36bb](https://github.com/solana-labs/solana/commit/1da36bba475fc63d9b2be272bcb056024161765f)

`create_executor()` is called in multiple places (in deploy, upgrade and finalize), why not cover these as well?
Or just measure the method, not the call sites?

> Do you think it's necessary to break down the `execute_time` here:
> 
> that covers the program execution into any further detail, i.e. more granularity in `execute_program_jit`?

On one validator it is either interpreter or JIT, so this would only make sense to compare across validators.

**Pull request closed without merging**